### PR TITLE
set up code formatting and linting tools

### LINKED ISSUE #74: 💡 [REQUEST] - Introduce a code formatter and linting tool
### Reference Issues

_No response_

### Summary

Currently there are various different code and indentation styles. To improve maintainability, we should use a unified format across all files. 

### Basic Example

Some examples of errors that would have been picked up by a lint tool: 
 - [invalid html tags](https://github.com/se310-t6/politicry/blob/bd67a08/backend/static/notfound.html#L11)
 - [undefined variable i](https://github.com/se310-t6/politicry/blob/1ecb1c4/extension/public/reddit.js#L65)

### Drawbacks

- more complicated to set up the development environment, since we would need to install 1 or 2 vscode extensions (prettier and/or eslint)

### Unresolved questions

_No response_

### PR DESCRIPTION
**Pull request by k-yle**
> **Note**
> This PR will be much easier to review if you look at the first commit separately. The second commit is just formatting and fixing some lint errors in `popup.js`

> **Note**
> To reduce frustration, GitHub Actions will accept any commits with formatting/lint issues (for now) until everyone has set up the VS Code exention and the system is working smoothly.

## Todo

- [x] introduced a code formatting tool ([prettier](https://prettier.io))
- [x] set up the code formatting tool to run automatically when you save a file in VS Code
- [x] set up the code formatting tool to run in the CI
- [x] introduced a lint tool ([eslint](https://eslint.org) and sonarlint)
- [x] set up the lint tool to run automatically when you save a file in VS Code
- [x] set up the lint tool to run in the CI
- [ ] show eslint and prettier warnings in sonarcloud dashboard (this will be done in a separate issue: #75)

## Motivation

Inconsistent and confusing formatting mades code harder to maintain. Additionally, sonarlint & eslint will help us write more maintainable code by detecting code-smells as we type. See issue #74 for more details.

## Attached GitHub issue

Closes #74

### Documentation
- [x] Updated the readme documents (`README.md` etc.)
- [x] ~~New functions and methods have additional comments added to document their usage~~

### Local Build
- [x] Ran all pre-commit hooks `pre-commit run --all-files`
- [x] ~~Extension has been tested to build correctly `cd extension && yarn && yarn build`~~
- [x] ~~Backend has been tested to build correctly `cd backend && docker-compose build`~~
- [x] ~~Extension test suite has been run locally `cd extension && yarn && yarn test`~~
- [x] ~~Backend test suite has been run locally `cd backend && python3 -m unittest`~~

### GitHub
- [x] Target branch has been set correctly
- [x] PR has been rebased onto target branch
- [x] PR has been assigned an owner
- [x] Author has performed a self-review of the code
- [x] Removed the WIP message from the top of this PR

### CODE REVIEW

**Comment by k1dubv1** [comment]
Yup LGTM!

**Review by k1dubv1 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "devDependencies": {
+    "eslint": "^8.23.1",
+    "eslint-config-kyle": "^9.3.0",
```
Comment (Destiny-02) [inline package.json:4]: Just checking but should this have your name in it?

**Review by Kanelmis — APPROVED** [review]
Formatting looks good

File: package.json (hunk 1 above)
Comment (k-yle) [reply package.json:4]: I've re-used a preset to save time, will remove it once #75 is fixed

**Pull request merged**